dipy/viz/ui.py

```diff
@@ -0,0 +1,562 @@
+import numpy as np
+
+from dipy.viz.interactor import CustomInteractorStyle
+
+from dipy.utils.optpkg import optional_package
+
+# Allow import, but disable doctests if we don't have vtk.
+vtk, have_vtk, setup_module = optional_package('vtk')
+
+if have_vtk:
+    version = vtk.vtkVersion.GetVTKSourceVersion().split(' ')[-1]
+    major_version = vtk.vtkVersion.GetVTKMajorVersion()
+else:
+    vtkInteractorStyleUser = object
+
+
+class UI(object):
+    """ An umbrella class for all UI elements.
+    While adding UI elements to the renderer, we need to go over all the
+     sub-elements that come with it and add those to the renderer too.
+    There are several features that are common to all the UI elements:
+    - ui_param : This is an attribute that can be passed to the UI object
```

Comment:
Agreed. @ranveeraggarwal : Please take a look at the numpy docstring standard: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt